ranking/get_ranking.go

```diff
@@ -0,0 +1,71 @@
+package ranking
+
+import (
+	"github.com/PretendoNetwork/nex-go"
+	ranking "github.com/PretendoNetwork/nex-protocols-go/ranking"
+	ranking_types "github.com/PretendoNetwork/nex-protocols-go/ranking/types"
+)
+
+func getRanking(err error, client *nex.Client, callID uint32, rankingMode uint8, category uint32, orderParam *ranking_types.RankingOrderParam, uniqueID uint64, principalID uint32) uint32 {
+	if commonRankingProtocol.getRankingsAndCountByCategoryAndRankingOrderParamHandler == nil {
+		logger.Warning("Ranking::GetRanking missing GetRankingsAndCountByCategoryAndRankingOrderParamHandler!")
+		return nex.Errors.Core.NotImplemented
+	}
+	rmcResponse := nex.NewRMCResponse(ranking.ProtocolID, callID)
+	server := client.Server()
+
+	if err != nil {
+		logger.Error(err.Error())
+		rmcResponse.SetError(nex.Errors.Ranking.Unknown)
+	}
+
+	err, rankDataList, totalCount := commonRankingProtocol.getRankingsAndCountByCategoryAndRankingOrderParamHandler(category, orderParam)
+	if err != nil {
+		logger.Error(err.Error())
+		rmcResponse.SetError(nex.Errors.Ranking.Unknown)
+	}
+
+	if totalCount == 0 || len(rankDataList) == 0 {
+		rmcResponse.SetError(nex.Errors.Ranking.NotFound)
+	}
+
+	if err == nil && totalCount != 0 {
```

Comment:
With Dani's suggestion to change lines 25 and 29 into returns, this check would no longer be needed so long as the check on line 28 is split into 2 checks

Reply:
I'm a bit confused by "so long as the check on line 28 is split into 2 checks" :p